README.md

```diff
@@ -1,12 +1,114 @@
 
-Google Enterprise Certificate Proxies are part of the [Google Cloud Zero Trust architecture][zerotrust] that enables mutual authentication with [client-side certificates][clientcert]. This repository contains a set of proxies/modules that can be used by clients or toolings to interact with certificates that are stored in [protected key storage systems][keystore].
+Google Enterprise Certificate Proxies (ECP) are part of the [Google Cloud Zero Trust architecture][zerotrust] that enables mutual authentication with [client-side certificates][clientcert]. This repository contains a set of proxies/modules that can be used by clients or toolings to interact with certificates that are stored in [protected key storage systems][keystore].
 
 To interact the client certificates, application code should not need to use most of these proxies within this repository directly. Instead, the application should leverage the clients and toolings provided by Google such as [Cloud SDK](https://cloud.google.com/sdk) to have a more convenient developer experience.
 
-## Installation
+## Compatibility
 
-TBD
+Currently ECP is in Preview stage and all the APIs and configurations are **subject to change**.
+
+Tthe following platforms are supported by ECP:
+
+- MacOS (Keychain, PKCS#11)
+- Windows (MyStore, PKCS#11)
+- Linux (PKCS#11 only)
+
+## Quick Start Guide
+
+### Prerequisites
+
+Before using ECP with your application/client, you should follow the instructions [here](enterprisecert) to configure your enterprise certificate policies with Access Context Manager. 
+
+### Installation
+
+1. Install Openssl
+`brew install openssl@1.1`
+
+1. Install gcloud CLI (Cloud SDK) at: https://cloud.google.com/sdk/docs/install
+
+1. Download the ECP binary based on your OS from the latest [Github release](https://github.com/googleapis/enterprise-certificate-proxy/releases).
+
+1. Unzip the downloaded zip and move all the binaries into the following directory:
+```
+<gcloud-installation-path>/google-cloud-sdk/bin/ecp/
+```
+
+1. If using gcloud’s bundled Python, skip to the next step. If not, install pyopenssl==22.0.0 and cryptography==36.0.2
+```
+pip install cryptography==36.0.2
+pip install pyopenssl==22.0.0
+```
+
+1. Create a new JSON file at `.config/gcloud/certificate_config.json` or you can put the json in the location of your choice and set the path to it using:
+```
+gcloud config set context_aware/auto_discovery_file_path "<json file path>"
+```
+
+1. Update the `certificate_config.json` file with details about the certificate (See [Configuration](#configutation) section for details.)
+
+1. Enable usage of client certificates through gcloud CLI config command:
+```
+gcloud config set context_aware/use_client_certificate true
+```
+
+### Configuration
+
+ECP relies on the `certificate_config.json` file to read all the metadata information of locating the certificate. The contents of this JSON file looks like the following:
+
+#### MacOS (Keychain)
```

Comment:
Please update all the config examples based on Carl's config design doc. We hope to get the config change in before public preview.

Reply:
Done.